### PR TITLE
Move coverity to a reusable workflow, add a dispatcher workflow

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

I need to find a way to test this, but review and comments welcome in the meantime :)

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: .github/workflows/misc-dailies.yml [hunk 1]
```diff
@@ -42,99 +42,29 @@ jobs:
...
-      - run: inv install-clang
-      - run: inv install-auth-build-deps
-      - run: inv install-coverity-tools PowerDNS
-      - run: inv coverity-clang-configure
-      - run: inv ci-autoconf
-      - run: inv ci-auth-configure
-      - run: inv coverity-make
-      - run: inv coverity-tarball auth.tar.bz2
-      - run: inv coverity-upload ${{ secrets.coverity_email }} PowerDNS auth.tar.bz2
+    uses: PowerDNS/pdns/.github/workflows/coverity.yml@master
```
Comment (romeroalx) [inline .github/workflows/misc-dailies.yml:45]: LGTM!

The only thing I am missing is that this change will require adding the permission `actions: read` for this workflow

Comment (rgacogne) [reply .github/workflows/misc-dailies.yml:48]: Thanks, it's always very hard for me to get that right without trial and error.. Do you have a good resource on the subject, by any chance?
I pushed a commit that hopefully should fix it, but please check it carefully :)

Comment (romeroalx) [reply .github/workflows/misc-dailies.yml:48]: Well... it turns out I was wrong. I did need this tag in the past but it seems it is not required anymore (I tested it)

Also, the documentation says it is only required for private repositories [https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#get-a-workflow](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#get-a-workflow)

From what I tested, it is safe to remove that permission

Comment (rgacogne) [reply .github/workflows/misc-dailies.yml:48]: Ah, I tried hard to find that information but did not manage to, well done! :) I'll remove the last commit then, worst case scenario Coverity scans will be broken for a day.

**Review by romeroalx — APPROVED** [review]
LGTM!

You may see some tests below:

- Test dispatch (do not consider the title of the run): [https://github.com/romeroalx/pdns/actions/runs/10954892376](https://github.com/romeroalx/pdns/actions/runs/10954892376)

- Test misc-dailies: [https://github.com/romeroalx/pdns/actions/runs/10954561859](https://github.com/romeroalx/pdns/actions/runs/10954561859)

File: .github/workflows/misc-dailies.yml (hunk 1 above)
Comment (rgacogne) [reply .github/workflows/misc-dailies.yml:45]: Removed!

**Comment by rgacogne** [comment]
Wonderful, thank you for testing!

**Pull request merged**